src/v/kafka/server/connection_context.cc

```diff
@@ -571,6 +603,47 @@ connection_context::dispatch_method_once(request_header hdr, size_t size) {
     auto self = shared_from_this();
     auto rctx = request_context(
       self, std::move(hdr), std::move(buf), sres->backpressure_delay);
+
+    if (!_is_virtualized_connection || !rctx.header().client_id.has_value()) {
+        co_return co_await _protocol_state.process_request(
+          shared_from_this(), std::move(rctx), sres);
+    }
+    auto v_connection_id = parse_virtual_connection_id(
+      rctx.header().client_id_buffer);
```

Comment:
question: Is there any requirement/need to handle authentication messages differently than other requests?  Like, only handle authn requests if it has the magic `mpx` header value?  As RP now supports re-authentication should that be disallowed in a virtual connection?  How does the MPX system authenticate with Redpanda?  Is it via SASL/SCRAM or will it use mTLS?

Reply:
Maybe worth noting that reauth is performed automagically (transparently) by modern enough Kafka clients. If the MPX system needs to mediate requests in some way to conform to the virtual connection protocol (and if reauthentication is desirable), this might require additional handling/care.

Reply:
i went over the code and it seems that there is no additional handling required for authentication/reauth requests. They do not have any special handling of client id. I think in this case we should relay on client to order the requests correctly.

Reply:
> They do not have any special handling of client id

The question I was trying to pose was whether or not authentication for virtual connections should be handled differently based on client ID (e.g. only honor auth requests from the `mpx` magic client for example).

> I think in this case we should relay on client to order the requests correctly.

I think that's fine.  I'm just posing the question as any change of authentication will effect all virtual connections.